From 0570730c16307a72f8241df12363f76600baf57d Mon Sep 17 00:00:00 2001
From: Edward Adam Davis <eadavis@qq.com>
Date: Fri, 21 Jun 2024 10:14:10 +0800
Subject: [PATCH] hfsplus: fix uninit-value in copy_name

syzbot reported a uninit-value in copy_name:
=====================================================
BUG: KMSAN: uninit-value in sized_strscpy+0xc0/0x150
copy_name+0x1c6/0x240 fs/hfsplus/xattr.c:414
hfsplus_listxattr+0x9e8/0x1330 fs/hfsplus/xattr.c:752

Use kzalloc for strbuf to avoid uninitialized value.

Reported-by: syzbot+e9ebc74caab161e1eb04@syzkaller.appspotmail.com
Signed-off-by: Edward Adam Davis <eadavis@qq.com>
Signed-off-by: Christian Brauner <brauner@kernel.org>
---
 fs/hfsplus/xattr.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/fs/hfsplus/xattr.c b/fs/hfsplus/xattr.c
index 2cf847d92..cae5cf1c3 100644
--- a/fs/hfsplus/xattr.c
+++ b/fs/hfsplus/xattr.c
@@ -700,7 +700,7 @@ ssize_t hfsplus_listxattr(struct dentry *dentry, char *buffer, size_t size)
 		return err;
 	}
 
-	strbuf = kmalloc(NLS_MAX_CHARSET_SIZE * HFSPLUS_ATTR_MAX_STRLEN +
+	strbuf = kzalloc(NLS_MAX_CHARSET_SIZE * HFSPLUS_ATTR_MAX_STRLEN +
 			XATTR_MAC_OSX_PREFIX_LEN + 1, GFP_KERNEL);
 	if (!strbuf) {
 		res = -ENOMEM;
-- 
2.43.0
